packages/devtools_app/lib/src/memory/flutter/memory_analyzer.dart

```diff
@@ -166,107 +165,295 @@ void imageAnalysis(
+
+        final imageCacheNode = processMatches(controller, matches);
+        if (imageCacheNode != null) {
+          librariesNode.addChild(imageCacheNode);
+        }
+
         analysisSnapshot.addChild(librariesNode);
+    }
+  });
+}
+
+AnalysisReference processMatches(
+  MemoryController controller,
+  Map<String, List<String>> matches,
+) {
+  // Root __FIELDS__ is a container for children, the children
+  // are added, later, to a treenode - if the treenode should
+  // be created.
+  final AnalysisField pending = AnalysisField(
+    '__FIELDS__',
+    null,
+  );
+  final AnalysisField cache = AnalysisField(
+    '__FIELDS__',
+    null,
+  );
+  final AnalysisField live = AnalysisField(
+    '__FIELDS__',
+    null,
+  );
+
+  var countPending = 0;
+  var countCache = 0;
+  var countLive = 0;
+  bool imageCacheFound = false;
+  matches.forEach((key, values) {
+    final fields = key.split('.');
+    imageCacheFound = fields[0] == imageCache;
+
+    for (final value in values) {
+      switch (fields[1]) {
+        case '_pendingImages':
+          countPending++;
+          pending.addChild(AnalysisField('url', value));
+          break;
+        case '_cache':
+          countCache++;
+          cache.addChild(AnalysisField('url', value));
+          break;
+        case '_liveImages':
+          countLive++;
+          live.addChild(AnalysisField('url', value));
+          break;
+      }
+    }
+  });
 
-        for (final ClassReference ref in value) {
-          // Insure instances are realized (not Reference.empty).
-          computeInstanceForClassReference(controller, ref);
-          final HeapGraphClassLive liveClass = ref.actualClass;
+  if (imageCacheFound) {
+    final imageCacheNode = AnalysisReference(
+      imageCache,
+      countNote: countPending + countCache + countLive,
+    );
 
-          final objectNode = AnalysisReference(
-            '${ref.name}',
-            countNote: liveClass.instancesCount,
+    final pendingNode = AnalysisReference(
+      'Pending',
+      countNote: countPending,
+    );
+
+    final cacheNode = AnalysisReference(
+      'Cache',
+      countNote: countCache,
+    );
+
+    final liveNode = AnalysisReference(
+      'Live',
+      countNote: countLive,
+    );
+
+    final pendingInstance = AnalysisInstance(
+      controller,
+      'Images',
+      pending,
+    );
+    final cacheInstance = AnalysisInstance(
+      controller,
+      'Images',
+      cache,
+    );
+    final liveInstance = AnalysisInstance(
+      controller,
+      'Images',
+      live,
+    );
+
+    pendingNode.addChild(pendingInstance);
+    imageCacheNode.addChild(pendingNode);
+
+    cacheNode.addChild(cacheInstance);
+    imageCacheNode.addChild(cacheNode);
+
+    liveNode.addChild(liveInstance);
+    imageCacheNode.addChild(liveNode);
+
+    return imageCacheNode;
+  }
+
+  return null;
+}
+
+// Flag to monitor field/object drill in.
+String monitorDrillIn = '';
```

Comment:
to follow the flutter scheme consider also prefixing this field with the word _debug
to make it clear it is debugging only.

Reply:
Done.